### PR TITLE
Remove cdi check webhook to be aligned with k8s eventual consistency

### PR DESCRIPTION
**Pull request by ShellyKa13**
**What this PR does / why we need it**:
    We should prevent rejecting a VM from being created
    if it can eventually be valid.
    It is possible CDI is on the process of being installed
    and DV resource will be avaliable soon, we should allow the
    VM to be created in such case.
    The VM will show in the status conditions that the dv resource
    can't be found. Once CDI is installed the VM will eventually start.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes # Jira-ticket: https://issues.redhat.com/browse/CNV-35866

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
None
```

### CODE REVIEW

File: pkg/virt-controller/watch/vm_test.go
Comment (xpivarc) [inline pkg/virt-controller/watch/vm_test.go:332]: Why is this important?

File: pkg/virt-controller/watch/vm_test.go
Comment (xpivarc) [inline pkg/virt-controller/watch/vm_test.go:337]: Could we use a fake client?

File: tests/operator/operator.go
Comment (xpivarc) [inline tests/operator/operator.go:2410]: Isn't this still relevant?

**Review by xpivarc — CHANGES_REQUESTED** [review]

File: pkg/virt-controller/watch/vm_test.go
Comment (ShellyKa13) [reply pkg/virt-controller/watch/vm_test.go:337]: what do you mean?

File: tests/operator/operator.go
Comment (ShellyKa13) [reply tests/operator/operator.go:2410]: No, this test checks that we fail to create vm if cdi doesnt exist, we removed the webhook so it shouldn't fail.
More over this is a test that shouldn't have existed, its very time and resource consuming for something that a UT can cover quit simply.

File: pkg/virt-controller/watch/vm_test.go
Comment (ShellyKa13) [reply pkg/virt-controller/watch/vm_test.go:332]: not important, ill remove

File: tests/operator/operator.go
Comment (xpivarc) [reply tests/operator/operator.go:2410]: Right, it should not fail but we should test the new expected behavior. I am ok if we have a unit test.

File: pkg/virt-controller/watch/vm_test.go
Comment (xpivarc) [reply pkg/virt-controller/watch/vm_test.go:337]: This is mock, I was suggesting using fake as we do for cdi client. But that might be more work.

Comment (ShellyKa13) [reply pkg/virt-controller/watch/vm_test.go:336]: I did the same as done in other tests here. works good :)

File: pkg/virt-controller/watch/vm_test.go
Comment (ShellyKa13) [reply pkg/virt-controller/watch/vm_test.go]: @xpivarc I remove this.

File: pkg/virt-controller/watch/vm_test.go
Comment (akalenyu) [inline pkg/virt-controller/watch/vm_test.go:229]: Is this what the "dummyinformer" returns in case CDI is not installed?

Comment (ShellyKa13) [reply pkg/virt-controller/watch/vm_test.go:229]: yep I ran the situation manually

Comment (xpivarc) [reply pkg/virt-controller/watch/vm_test.go:229]: This is not `dummyInformer`, this is the real client that is always there.

Comment (akalenyu) [reply pkg/virt-controller/watch/vm_test.go:229]: Yeah I just wanted to know this simulates the real thing, where if CDI doesn't exist the dummyinformer is used

File: tests/operator/operator.go
Comment (ShellyKa13) [reply tests/operator/operator.go:2410]: @xpivarc if I convinced you i'll appreciate lgtm :)

**Review by xpivarc — COMMENTED** [review]
/approve

**Comment by oshoval** [comment]
/lgtm
/hold

just testing something, will remove both in a moment

**Comment by ShellyKa13** [comment]
@oshoval Hi can you explain what you were checking and why you removed the lgtm eventually?

**Comment by oshoval** [comment]
> @oshoval Hi can you explain what you were checking and why you removed the lgtm eventually?

I was checking a plugin of prow, not related at all to this PR
it just monitors adding labels
i didnt review the PR, this is why i removed my lgtm, and used hold so it wont be merged by mistake

please disregard, as i just reverted my effect
i just wanted to see if the plugin comment what it should
https://github.com/kubevirt/kubevirt/pull/11010#issuecomment-1894030962
once the plugin detect lgtm + approve it runs few additional jobs that will run only when lgtm + approve present
in order to save CI resources

**Pull request merged**
